astlib/history.mli

```diff
@@ -27,8 +27,8 @@ val convert
 
 type 'node conversion_function
   = 'node Ast.t
-  -> to_ast:('node -> version:Version.t -> 'node Ast.t)
-  -> of_ast:('node Ast.t -> version:Version.t -> 'node)
+  -> to_src_ast:('node -> 'node Ast.t)
```

Comment:
I'm not sure what the best terminology is here but I find those name a bit confusing.

What helped me writing the migration functions was seeing them as a deconstructor and a builder of some sort. In other words, you use `to_src_ast` to deconstruct `src` values and pattern match over them and use `of_dst_ast` to build new `dst` values. What would you think about reusing that terminology here?

Reply:
I think overall it's okay because the types won't let you misuse those functions anyway.

I just find it particularly hard because the `to`/`of` prefixes are swapped compared to my mental model of what those functions do.

Reply:
I pushed another commit naming things as `wrap` / `unwrap`, I think it comes out much better. In the upgrade/downgrade implementation they're a bit wordy, `unwrap_src_version` and `wrap_dst_version`,  I erred on the side of clarity. What do you think?

Reply:
Or should those just be `wrap` and `unwrap` as well, with the versions being implied?

Reply:
Never mind, I went ahead and made them all `wrap` and `unwrap`.